CONTRIBUTING.md

```diff
@@ -264,6 +264,8 @@ This project follows the [Kubernetes API conventions](https://github.com/kuberne
 
 * Status fields MUST be optional. Our controllers are patching selected fields instead of updating the entire status in every reconciliation.
 
+* If a field is required (for our controllers to work) and has a default value, but we don't want to force users to set the field, we have to mark the field as optional. Otherwise, the client-side kubectl OpenAPI schema validation will force the user to set it even though it would be defaulted on the server-side.
```

Comment:
Am  wrong or this applies only of the default value we want to use is different than the default value for a given type (e.g int that should default to 1 instead of 0)?

Reply:
Yes, default value in this context means a default value specified via OpenAPI schema. Problem is that:
* on client-side (kubectl per default): the validation is applied
* on server-side: first the defaulting and then the validation is applied

Probably could do make that a bit clearer.

Reply:
Updated to clarify, let me know if it's still not clear enough.